test/e2e/snaps/test-snap-txinsights.spec.js

```diff
@@ -0,0 +1,124 @@
+const { strict: assert } = require('assert');
+const { withFixtures } = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+const { TEST_SNAPS_WEBSITE_URL } = require('./enums');
+
+describe('Test Snap TxInsights', function () {
+  it('tests tx insights functionality', async function () {
+    const ganacheOptions = {
+      accounts: [
+        {
+          secretKey:
+            '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        },
+      ],
+    };
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions,
+        failOnConsoleError: false,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+
+        // enter pw into extension
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // navigate to test snaps page and connect
+        await driver.driver.get(TEST_SNAPS_WEBSITE_URL);
+        await driver.delay(1000);
+
+        // find and scroll to the bip32 test and connect
+        const snapButton1 = await driver.findElement('#connectInsightsSnap');
+        await driver.scrollToElement(snapButton1);
+        await driver.delay(1000);
+        await driver.clickElement('#connectInsightsSnap');
```

Comment:
[OPTIONAL] consider using `data-testid` for targeting DOM elements instead of relying on ids.

Reply:
The id tags referenced in the E2E tests are directly referring to id tags inside the test-snaps page. It was specifically designed by us to be used for our E2E testing (similar to test-dapp). 
A request to move from class to id was made by Erik Marks (https://github.com/MetaMask/test-snaps/issues/42) and we chose to use IDs from then on, also in the newest iteration of the react test-snaps page (https://metamask.github.io/test-snaps/latest) 
If there is a very compelling argument to refactor the test-snaps page again - it could be considered, we have just had no reason to so far as it would require changing all E2Es and the test-snaps page itself.